### PR TITLE
Implement product view

### PR DESCRIPTION
**Pull request by zebasAL**
-I did a product view for all the homepage list
-it has an interactive css 

@chiItepin 

image
image

### CODE REVIEW

File: src/components/ui/ProductDetails.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          <h2 data-cy="product-info-title" className="product-info-title">{productvalues.title}</h2>
+          <p data-cy="product-info-price" className="product-info-price">
+            {`$${productvalues.price} USD`}
+          </p>
+          <form onSubmit={handleSubmit}>
+            <p>Quantity</p>
+            <QuantityBtn
+              validateNewQuantity={validateNewQuantity}
+              validateSubtraction={validateSubtraction}
+              validateSum={validateSum}
```
Comment (chiItepin) [inline src/components/ui/ProductDetails.jsx:53]: Remove unnecessary validations and instead, pass one value to the input field and pass a function that will handle setting the value if the value. entered is 0 or greater

File: src/components/ui/ProductDetails.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+              productQuntity={productQuntity}
```
Comment (chiItepin) [inline src/components/ui/ProductDetails.jsx:54]: Fix typo

File: src/components/ui/ProductDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+              setProductQuntity={setProductQuntity}
```
Comment (chiItepin) [inline src/components/ui/ProductDetails.jsx:55]: Fix typo

File: src/containers/ProductsView.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import { useParams } from 'react-router-dom';
+import { toaster, Spinner } from 'evergreen-ui/';
+import ProductDetails from '../components/ui/ProductDetails';
+
+const ProductsView = () => {
+  const [product, setProduct] = useState({}); // [{}, () => {}]
+  const [loaded, setLoaded] = useState(false);
+  const { id } = useParams(); // { category: 'jewelry', id: 1}
+
+  const getProductid = () => {
```
Comment (chiItepin) [inline src/containers/ProductsView.jsx:12]: Rename it to `getProduct`

File: src/containers/ProductsView.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  if (loaded === false) return <div style={{ display: 'table', margin: '10px auto' }}><Spinner /></div>;
+
+  return (
+    <div>
+      <ProductDetails productvalues={product} />
+    </div>
+  );
+};
+
+export default ProductsView;
```
Comment (chiItepin) [inline src/containers/ProductsView.jsx:37]: Rename it to `ProductView` because only one product will be shown

File: src/mobile.css [hunk 6]
```diff
@@ -1,10 +1,11 @@
 
-@media only screen and (max-width: 999px) {
+ @media only screen and (min-width: 200px) and (max-width: 999px)  {
```
Comment (chiItepin) [inline src/mobile.css:2]: Remove unreachable min-width

File: src/components/ui/QuantityBtn.jsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  validateSubtraction,
+  validateSum,
+  validateNewQuantity,
+}) => (
+  <div className="quantity-btn">
+    <Button data-cy="substract-quantity-btn" onClick={validateSubtraction} borderRadius="0" borderRight="none">-</Button>
+    <TextInput
+      id="quantity-btn"
+      type="number"
+      value={productQuntity}
```
Comment (chiItepin) [inline src/components/ui/QuantityBtn.jsx:16]: fix typo

File: src/components/ui/ProductDetails.jsx [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+          <p data-cy="product-info-description" className="product-info-description">{productvalues.description}</p>
+          <CollapseBtns productDetails={productvalues} />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+ProductDetails.propTypes = {
+  productvalues: PropTypes.shape({
```
Comment (chiItepin) [inline src/components/ui/ProductDetails.jsx:71]: should be in camelCase

File: src/components/ui/ProductDetails.jsx [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    title: PropTypes.string,
+    price: PropTypes.number,
+    description: PropTypes.string,
+    category: PropTypes.string,
+    image: PropTypes.string,
+    rating: PropTypes.objectOf(PropTypes.number).isRequired,
```
Comment (chiItepin) [inline src/components/ui/ProductDetails.jsx:77]: This is wrong, should use `shape` method to describe each property type:
`    rating: PropTypes.shape({
      rate: PropTypes.number,
      count: PropTypes.number,
    }),`

**Review by chiItepin — CHANGES_REQUESTED** [review]

File: src/components/ui/ProductDetails.jsx (hunk 1 above)
Comment (zebasAL) [reply src/components/ui/ProductDetails.jsx:53]: image

File: src/components/ui/ProductDetails.jsx (hunk 2 above)
Comment (zebasAL) [reply src/components/ui/ProductDetails.jsx:54]: fixed above image

File: src/containers/ProductsView.jsx (hunk 4 above)
Comment (zebasAL) [reply src/containers/ProductsView.jsx:12]: image
Const renamed

File: src/containers/ProductsView.jsx (hunk 5 above)
Comment (zebasAL) [reply src/containers/ProductsView.jsx:37]: component and all modules were renamed

File: src/mobile.css (hunk 6 above)
Comment (zebasAL) [reply src/mobile.css:2]: image

File: src/components/ui/ProductDetails.jsx (hunk 8 above)
Comment (zebasAL) [reply src/components/ui/ProductDetails.jsx:71]: image
 all productValues props were changed

File: src/components/ui/QuantityBtn.jsx (hunk 7 above)
Comment (zebasAL) [reply src/components/ui/QuantityBtn.jsx:16]: image

File: src/components/ui/ProductDetails.jsx (hunk 9 above)
Comment (zebasAL) [reply src/components/ui/ProductDetails.jsx:77]: image

**Review by chiItepin — APPROVED** [review]

**Pull request merged**